### ISSUE TITLE
Deployment error - Error: The resource you are looking for has been removed, had its name changed...

### ISSUE THREAD

**Issue by jon64digital** [Microsoft/vscode-azureappservice#395]
Repro steps:
Trying to deploy to Azure using Azure App Service Extension. I have done this many times but it suddenly stopped working. I have tried signing out and signing in again. Other apps deply fine, it's the only one in my list of apps that gets this error when it deploys.

The site hasn't moved, it's still there in Azure. I can see it in the portal and I can browse the site. I did add a custom domain but I'm fairly sure it was still working after that.

Action: appService.Deploy
Error type: Error
Error Message: The resource you are looking for has been removed, had its name changed, or is temporarily unavailable.

Version: 0.7.0
OS: win32

**Comment by nturinski** [comment]
Hi @jon64digital,

We've been having receiving issues about sudden deployment failures.  It would be awesome if you could help us pinpoint the problem by trying to deploy using one of two methods:

1. If you have the Azure CLI installed, could you try to deploy to your web app using the following command:
`az webapp deployment source config-zip --name
                                       --resource-group
                                       --src
                                       [--slot]`
(reference: https://docs.microsoft.com/en-us/cli/azure/webapp/deployment/source?view=azure-cli-latest#az-webapp-deployment-source-config-zip)

2.  If you could install a previous version of the Azure App Service Extension by manually installing this .vsix (https://github.com/Microsoft/vscode-azureappservice/releases/download/v0.6.1/vscode-azureappservice-0.6.1.vsix) and then attempt a deploy, it would tell us if we made breaking changes in v0.7.0 or if the changes were introduced somewhere else.

If you need to update your web app immediately, please try to use the  Local Git Deployment Source, or connect it to a GitHub repository.  This can be done in the "Configure Deployment Source" context action.  Sorry for the inconvenience.

**Comment by jon64digital** [comment]
I don't have the CLI installed but I will try to do that tonight or tomottow morning. I'll also try the older extension.

**Comment by jon64digital** [comment]
OK, so I just downloaded the CLI and worked it out. I managed to do a single deployment but forgot to include the node_modules folder in the zip so got a 500 error. Next time I tried I got the following error in the CLI:

`Zip deployment https://myAppName.scm.azurewebsites.net/api/zipdeploy failed with status code '500' and reason '<html><head><title>500 - The request timed out.</title></head><body>  <font color ="#aa0000">         <h2>500 - The request timed out.</h2></font>  The web server failed to respond within the specified time.</body></html>'`

So, I thought I'd try it once more and got the following error:

`Zip deployment https://jonexperiment.scm.azurewebsites.net/api/zipdeploy failed with status code '409' and reason 'There is a deployment currently in progress. Please try again when it completes.'`

Then, a few minutes later it finally finished and gave me a success error (it was still running despite the timeout).

Long story short, it published using the CLI but I thought you might want to know the CLI seems buggy. It takes a lot longer then the VScode extension and gives no feedback while deploying. I nearly gave up on it.

If you still need me to try the old version of the extension I can do that tomorrow.

**Comment by vikian050194** [comment]
I have the same problem. If some testing required, I would be happy to help with it.

**Comment by nturinski** [comment]
@jon64digital Thanks a lot for testing that out for me and being very descriptive-- it helped shed a lot of light on the situation.  I was able to repro a similar error using a custom domain so now we know where to start resolving this issue.

@vikian050194  I appreciate the offer, but I think we're good now!

**Comment by nturinski** [comment]
@jon64digital  @vikian050194  We just released a new version of the extension that should fix this issue.  Please give it a try and let me know if there are still any problems deploying.

Thanks again for your help in resolving this issue!

**Comment by jon64digital** [comment]
So it was the custom domain throwig it?

Seems to work now anyway. Many thanks.

**Comment by nturinski** [comment]
@jon64digital Yeah, we changed a hard-coded scm url to dynamically retrieving it from an array of hostnames in the last version update.  But a custom domain would change the index of where the scm url existed, causing the error.
